packages/elements/src/clock/index.ts

```diff
@@ -482,28 +526,58 @@ export class Clock extends ResponsiveElement {
   * @param event Event Object
   * @returns {void}
   */
-  private handleUpKey (event: KeyboardEvent): void {
-    this.shift(UP, this.getShiftAmountFromTarget(event.target));
+  private handleUpKey (): void {
+    const selection = this.renderRoot.querySelector(`[part~=${this.activeSegment}]`);
+    this.shift(Direction.UP, this.getShiftAmountFromTarget(selection));
```

Comment:
This code needs to be refactored to remove querySelector and `getShiftAmountFromTarget` function. You already know what is the `activeSegment` and can calculate `shiftAmount` from the active segment.

Reply:
Resolved by your PR